### PR TITLE
Feature/timetable controller

### PR DESCRIPTION
**Pull request by OlegAndrus12**

### CODE REVIEW

File: Core/DTO/Timetable.cs [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+
+namespace Core.DTO
+{
+    public class Timetable
+    {
+        public int Id { get; set; }
+		public string Day { get; set; } 
+		public string Period { get; set; }
+        public virtual Time Time { get; set; }
+        public int TimeID { get; set; }
```
Comment (nikilark) [inline Core/DTO/Timetable.cs]: Do you really need both id and item?

File: WebApplication/Views/Timetable/Timetable.cshtml [hunk 2]
```diff
@@ -0,0 +1,255 @@
+﻿@model Dictionary<int, List<Core.DTO.Timetable>>
```
Comment (nikilark) [inline WebApplication/Views/Timetable/Timetable.cshtml:1]: Why not use a specialized model?

File: WebApplication/Views/Timetable/Timetable.cshtml [hunk 3]
```diff
@@ -0,0 +1,255 @@
...
+                                }
+                            }
+                        }
+                        else
+                        {
+                            for (int j = 0; j < 5; ++j)
+                            {
+                                <div class="border timetable-item">
+                                    <a href="#myModal" data-backdrop="false" data-toggle="modal">
+                                        <p>Nothing</p>
```
Comment (nikilark) [inline WebApplication/Views/Timetable/Timetable.cshtml]: "Nothing" unnecessary

File: WebApplication/Views/Timetable/Timetable.cshtml [hunk 4]
```diff
@@ -0,0 +1,255 @@
...
+                                    </div>
+                                }
+                            }
+                        }
+                        else
+                        {
+                            for (int j = 0; j < 5; ++j)
+                            {
+                                <div class="border timetable-item">
+                                    <a href="#myModal" data-backdrop="false" data-toggle="modal">
```
Comment (nikilark) [inline WebApplication/Views/Timetable/Timetable.cshtml]: What notes will show "nothing"?

File: WebApplication/Views/Timetable/Timetable.cshtml [hunk 5]
```diff
@@ -0,0 +1,255 @@
...
+                            dragY = e.clientY - $(this).position().top;
+                        },
+                        mouseup: function () { dragging = false; },
+                        mousemove: function (e) {
+                            if (dragging)
+                                $(this).offset({ top: e.clientY - dragY, left: e.clientX - dragX });
+
+                        }
+                    })
+                })
```
Comment (nikilark) [inline WebApplication/Views/Timetable/Timetable.cshtml]: We have file for js

File: WebApplication/Controllers/TimetableController.cs [hunk 6]
```diff
@@ -0,0 +1,69 @@
...
+                .ToDictionary(x => x.Key, x => x.OrderBy(x => (int)((WeekDays)Enum.Parse(typeof(WeekDays), x.Day))).ToList())
+                ;
+            keyValues = keyValues.OrderBy(k => k.Key).ToDictionary(z => z.Key, y => y.Value);
+
+            foreach(var i in keyValues.Values)
+            {
+
+                for(int j = 0; j<i.Count; ++j)
+                {
+                    if ((int)((WeekDays)Enum.Parse(typeof(WeekDays), i[j].Day)) != j)
```
Comment (nikilark) [inline WebApplication/Controllers/TimetableController.cs]: This could be simplier

File: WebApplication/Controllers/TimetableController.cs [hunk 7]
```diff
@@ -0,0 +1,69 @@
...
+        private readonly ILogger<TimetableController> _logger;
+
+        public TimetableController(IMapper mapper, ITimeTableService timeTable, ILogger<TimetableController> logger)
+        {
+            _mapper = mapper;
+            _timeTable = timeTable;
+            _logger = logger;
+        }
+
+        public IActionResult Index()
```
Comment (nikilark) [inline WebApplication/Controllers/TimetableController.cs:26]: You used linq, foreach, for and while in one method. Please choose one style
What is more, think about reducing complexity of this method

File: Core/DTO/Time.cs [hunk 8]
```diff
@@ -0,0 +1,11 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Text;
+
+namespace Core.DTO
```
Comment (arihilmir) [inline Core/DTO/Time.cs]: Why do you need empty DTOs?

File: WebApplication/Controllers/TimetableController.cs [hunk 9]
```diff
@@ -0,0 +1,69 @@
...
+using System.Collections.Generic;
+using Services.TimeTableService;
+using WebApplication.Models;
+using System.Linq;
+using System;
+
+public enum WeekDays
+{
+    Monday = 0,
+    Tuesday = 1,
```
Comment (arihilmir) [inline WebApplication/Controllers/TimetableController.cs]: Unnecessary assignments. It is default behavior.

File: WebApplication/Views/Timetable/Timetable.cshtml [hunk 10]
```diff
@@ -0,0 +1,255 @@
...
+<head>
+    <meta charset="utf-8">
+    <meta name="viewport" content="width=device-width, initial-scale=1">
+    <link rel="stylesheet" href="https://stackpath.bootstrapcdn.com/bootstrap/4.4.1/css/bootstrap.min.css" integrity="sha384-Vkoo8x4CGsO3+Hhxv8T/Q5PaXtkKtu6ug5TOeNV6gBiFeWPGFN9MuhOf23Q9Ifjh" crossorigin="anonymous">
+    <link rel="stylesheet" href="~/css/timetableStyle.css">
+    <link href="https://cdnjs.cloudflare.com/ajax/libs/mdbootstrap/4.14.0/css/mdb.min.css" rel="stylesheet">
+    <script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.3.1/jquery.min.js"></script>
+</head>
+<body class="container-fluid">
+    <div class="content">
```
Comment (arihilmir) [inline WebApplication/Views/Timetable/Timetable.cshtml]: Where's navbar?

File: WebApplication/WebApplication.csproj [hunk 11]
```diff
@@ -12,6 +12,13 @@
 
     <ItemGroup>
       <PackageReference Include="AutoMapper" Version="9.0.0" />
+      <PackageReference Include="Microsoft.EntityFrameworkCore.SqlServer" Version="3.0.0" />
```
Comment (arihilmir) [inline WebApplication/WebApplication.csproj]: We don't use SqlServer in the project.

File: Core/DTO/Timetable.cs (hunk 1 above)
Comment (OlegAndrus12) [reply Core/DTO/Timetable.cs]: O`k, i will remove this.
Actually, i don`t understand difference between Entities and DTO

File: WebApplication/Views/Timetable/Timetable.cshtml (hunk 3 above)
Comment (OlegAndrus12) [reply WebApplication/Views/Timetable/Timetable.cshtml]: OK

File: WebApplication/Views/Timetable/Timetable.cshtml (hunk 4 above)
Comment (OlegAndrus12) [reply WebApplication/Views/Timetable/Timetable.cshtml]: First one fill the null items in the row,
Second one fill the whole row with "nothing" if such row doesn`t exist. 
It`s the case when user don`t lessons on this Time at all

File: WebApplication/Views/Timetable/Timetable.cshtml (hunk 5 above)
Comment (OlegAndrus12) [reply WebApplication/Views/Timetable/Timetable.cshtml]: ok

File: WebApplication/Controllers/TimetableController.cs (hunk 6 above)
Comment (OlegAndrus12) [reply WebApplication/Controllers/TimetableController.cs]: Maybe, i will add separated func

File: WebApplication/Controllers/TimetableController.cs (hunk 7 above)
Comment (OlegAndrus12) [reply WebApplication/Controllers/TimetableController.cs:26]: I`ll take it into consideration.
We should change html structure in order to reduce complexity. 
I`ll explain you this point on the Tuesday meeting

File: Core/DTO/Time.cs (hunk 8 above)
Comment (OlegAndrus12) [reply Core/DTO/Time.cs]: I decided not to fill them until i wil have some problems with them.
But fortunately, this extra DTO`s was uselless in my task and i simply forgot to delete them.
Could you explain where we have data integrity check (like FK)

File: WebApplication/Controllers/TimetableController.cs (hunk 9 above)
Comment (OlegAndrus12) [reply WebApplication/Controllers/TimetableController.cs]: Ok

File: WebApplication/Views/Timetable/Timetable.cshtml (hunk 10 above)
Comment (OlegAndrus12) [reply WebApplication/Views/Timetable/Timetable.cshtml]: It`s not my task.
Navbar shoud be created at Shared foulder and be accesed using Layout razor property on each page.

File: WebApplication/WebApplication.csproj (hunk 11 above)
Comment (OlegAndrus12) [reply WebApplication/WebApplication.csproj]: It`s VS generated one.
Do not consciously add this package

File: WebApplication/Views/Timetable/Timetable.cshtml (hunk 4 above)
Comment (nikilark) [reply WebApplication/Views/Timetable/Timetable.cshtml]: But times with "nothing" toggle modal with notes. That is why I asked, what notes will user see when he clicks on "Nothing"

Comment (OlegAndrus12) [reply WebApplication/Views/Timetable/Timetable.cshtml]: I got it.
Thanks

File: WebApplication/Views/Timetable/Timetable.cshtml (hunk 2 above)
Comment (OlegAndrus12) [reply WebApplication/Views/Timetable/Timetable.cshtml:1]: Why not this one?

File: WebApplication/Controllers/TimetableController.cs [hunk 12]
```diff
@@ -0,0 +1,71 @@
...
+        {
+            _mapper = mapper;
+            _timeTable = timeTable;
+            _logger = logger;
+        }
+
+        public IActionResult Index()
+        {
+            Dictionary<int, List<Core.DTO.Timetable>> keyValues;
+            Dictionary<string, int> daysOfWeek = new Dictionary<string, int>()
```
Comment (nikilark) [inline WebApplication/Controllers/TimetableController.cs:29]: const?

File: WebApplication/wwwroot/css/timetableStyle.css [hunk 13]
```diff
@@ -0,0 +1,303 @@
...
+}
+
+#logo {
+    height: 50px;
+}
+
+html, body, header {
+    height: 100%;
+    width: 100%;
+}
```
Comment (nikilark) [inline WebApplication/wwwroot/css/timetableStyle.css]: this and many below not touching timetable page. We`ve decided to make separate files, haven`t we?

File: WebApplication/Views/Timetable/Timetable.cshtml [hunk 14]
```diff
@@ -0,0 +1,234 @@
...
+                            <div class="row m-0 justify-content-between " id="timetable-notes-footer">
+                                <button class="btn btn-light m-2">Mark checked as done</button>
+                                <button class="btn btn-primary m-2">Ok</button>
+                            </div>
+                        </div>
+                    </div>
+                </div>
+            </div>
+            <script type="text/javascript" src ="~/js/timetableScript.js"></script>
+            <script src="https://cdnjs.cloudflare.com/ajax/libs/baguettebox.js/1.10.0/baguetteBox.min.js"></script>
```
Comment (nikilark) [inline WebApplication/Views/Timetable/Timetable.cshtml]: remove all, that should be in layout

File: WebApplication/Controllers/TimetableController.cs (hunk 12 above)
Comment (OlegAndrus12) [reply WebApplication/Controllers/TimetableController.cs:29]: _timeTable.LoadTimetableAsync().Result should return const value, else we will got an eror.
So, no const usage

File: WebApplication/wwwroot/css/timetableStyle.css (hunk 13 above)
Comment (OlegAndrus12) [reply WebApplication/wwwroot/css/timetableStyle.css]: You don`t mind if we write base CSS code in site.css ?

**Comment by OlegAndrus12** [comment]
+ Set as a convention in site.css we write only base CSS code. For separate pages we create separate css files. 
+ Tranfer links to layout.cshtml

File: WebApplication/Controllers/TimetableController.cs (hunk 12 above)
Comment (nikilark) [reply WebApplication/Controllers/TimetableController.cs:29]: daysOfWeek could be const

File: WebApplication/Views/Shared/_Layout.cshtml [hunk 15]
```diff
@@ -6,43 +6,23 @@
     <title>@ViewData["Title"] - WebApplication</title>
     <link rel="stylesheet" href="~/lib/bootstrap/dist/css/bootstrap.min.css" />
     <link rel="stylesheet" href="~/css/site.css" />
+    <link rel="stylesheet" href="https://stackpath.bootstrapcdn.com/bootstrap/4.4.1/css/bootstrap.min.css" integrity="sha384-Vkoo8x4CGsO3+Hhxv8T/Q5PaXtkKtu6ug5TOeNV6gBiFeWPGFN9MuhOf23Q9Ifjh" crossorigin="anonymous">
+    <link rel="stylesheet" href="~/css/timetableStyle.css">
```
Comment (nikilark) [inline WebApplication/Views/Shared/_Layout.cshtml]: it is shared layout, here should not be css file we will use only in one page

File: WebApplication/Views/Shared/_Layout.cshtml [hunk 16]
```diff
@@ -6,43 +6,23 @@
     <title>@ViewData["Title"] - WebApplication</title>
     <link rel="stylesheet" href="~/lib/bootstrap/dist/css/bootstrap.min.css" />
     <link rel="stylesheet" href="~/css/site.css" />
+    <link rel="stylesheet" href="https://stackpath.bootstrapcdn.com/bootstrap/4.4.1/css/bootstrap.min.css" integrity="sha384-Vkoo8x4CGsO3+Hhxv8T/Q5PaXtkKtu6ug5TOeNV6gBiFeWPGFN9MuhOf23Q9Ifjh" crossorigin="anonymous">
```
Comment (nikilark) [inline WebApplication/Views/Shared/_Layout.cshtml]: we have downloaded bootstrap in project. Does it worse than this?

File: WebApplication/wwwroot/css/site.css [hunk 17]
```diff
@@ -1,71 +1,72 @@
...
+  font: roboto;
+  font-size: 1.25rem;
+}
+
+.dropdown-menu {
+  min-width: 0px;
+}
+
+.dropdown-item {
+  padding: 4px;
```
Comment (nikilark) [inline WebApplication/wwwroot/css/site.css]: it was for news page

File: WebApplication/Views/Shared/_Layout.cshtml [hunk 18]
```diff
@@ -6,43 +6,23 @@
...
 
-    <footer class="border-top footer text-muted">
-        <div class="container">
-            &copy; 2020 - WebApplication - <a asp-area="" asp-controller="Home" asp-action="Privacy">Privacy</a>
-        </div>
-    </footer>
     <script src="~/lib/jquery/dist/jquery.min.js"></script>
     <script src="~/lib/bootstrap/dist/js/bootstrap.bundle.min.js"></script>
     <script src="~/js/site.js" asp-append-version="true"></script>
+    <script type="text/javascript" src ="~/js/timetableScript.js"></script>
```
Comment (nikilark) [inline WebApplication/Views/Shared/_Layout.cshtml]: same as for css

**Review by nikilark — APPROVED** [review]
All done

**Pull request merged**